default/python/AI/PolicyAI.py

```diff
@@ -336,19 +492,24 @@ def _get_infl_prod(self, update=False) -> float:
             fo.updateResourcePools()
         return self._empire.resourceProduction(fo.resourceType.influence)
 
-    def _can_adopt(self, name: str, replace_other: Optional[str] = None) -> bool:
-        """Note that when replace_other is set, this function currently assumes that with
+    def _can_adopt(self, name: str, replace: Union[str, Set[str], None] = None) -> bool:
+        """
+        Can we adopt named policy, possibly by replacing (one of) replace?
+        Note that when replace_other is set, this function currently assumes that with
         replace_other removed, there are no other exclusions for policy(name).
         If this won't do, the function needs to temporarily deadopt replace_other or
         policyPrereqsAndExclusionsOK must be extended so that replace_other can be passed to it.
         """
         if name in self._adopted:
             return False
         policy = fo.getPolicy(name)
+        wanted = self._wanted_ip * (name not in self._prioritised_policies)
```

Comment:
Boolean multiplication could be tricky.  

```suggestion
        wanted = self._wanted_ip if name in self._prioritised_policies else 0
```

Reply:
Actually it was more dangerous in dislike_multiplier than here.
Unless I am mistaken the problem is that something that looks like a bool is actually something else. But the result of a "not in" operator will always be a bool, won't it? Anyway, the alternative works fine here, too.

Reply:
It's not about technical stuff, it's about implementation and intention. 

`self._wanted_ip if name in self._prioritised_policies else 0` your intention is to define value by condition and implementation does exactly that. That is the simplest implementation.